dropwizard-core/src/main/java/io/dropwizard/setup/AdminEnvironment.java

```diff
@@ -59,11 +72,13 @@ public void addTask(Task task) {
     private void logTasks() {
         final StringBuilder stringBuilder = new StringBuilder(1024).append(String.format("%n%n"));
 
+
         for (Task task : tasks.getTasks()) {
+
             stringBuilder.append(String.format("    %-7s /tasks/%s (%s)%n",
                                                "POST",
                                                task.getName(),
-                                               task.getClass().getCanonicalName()));
+                                               getTaskClassName.apply(task)));
```

Comment:
I'd simply use [´MoreObjects#firstNonNull(T, T)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/MoreObjects.html#firstNonNull(T,%20T)) in this case. No need for this `Function<Task, String>`.